src/hub.js

```diff
@@ -464,8 +465,9 @@ document.addEventListener("DOMContentLoaded", async () => {
   hubPhxChannel.on("message", ({ session_id, type, body }) => {
     const userInfo = hubPhxPresence.state[session_id];
     if (!userInfo) return;
+    const maySpawn = scene.is("entered");
 
-    addToPresenceLog({ name: userInfo.metas[0].profile.displayName, type, body });
+    addToPresenceLog({ name: userInfo.metas[0].profile.displayName, type, body, maySpawn });
```

Comment:
Spawn state shouldn't really be a property of the message, it should probably be passed in from ui-root (which can get it from the scene, and know about changes with the `state-change` event). That way messages that were added before you joined still get buttons. Probably minor in practice, but feels a bit odd.

Reply:
ah this was more important in earlier versions where I was tagging the message as spawnable based upon a more complex expression (if the room has been entered, and you were the origin of the message, and I was presuming we would not have it spawnable if it was a photo eventually etc.) i'll leave it for now since it's a minor refactor if/when it starts breaking down